compliance/aws/long_stopped_instances/aws_long_stopped_instances.pt

```diff
@@ -0,0 +1,301 @@
+   res = []
+   var param_exclude_tags_lower=[];
+    for(var i=0; i < param_exclude_tags.length; i++){
+      param_exclude_tags_lower[i]=param_exclude_tags[i].toString().toLowerCase();
+    }
+    _.each(ds_ec2_stopped_instances_list, function(instance_stopped){
+      _.each(ds_cloudwatch_cpu_usage, function(cloudwatch_metrics){	  
+        var tags=instance_stopped['tags'];
+        var is_tag_matched=false
+        var tag_key_value=""
+        if(typeof(tags) != "undefined"){
+	      for(var k=0;k<tags.length;k++){
+            var tag=tags[k];
+            if((param_exclude_tags_lower.indexOf((tag['tagKey']).toLowerCase()) !== -1) ||      (param_exclude_tags_lower.indexOf((tag['tagKey']+'='+tag['tagValue']).toLowerCase()) !== -1)){
+                is_tag_matched = true;
+            }
+            // Constructing tags with comma separated to display in detail_template
+            if((tag['tagValue']).length > 0){
+              tag_key_value = tag_key_value + ', '+ tag['tagKey']+'='+tag['tagValue']
+            }else{
+              tag_key_value = tag_key_value + ', '+ tag['tagKey']
+            }
+		  }
+        }
+	    //If the EC2 instance tag does not match with entered param_exclude_tags and cloud metrics is empty,
+		if(instance_stopped["instance_id"] == cloudwatch_metrics["instance_id"] && cloudwatch_metrics['datapoints'].length == 0 && !is_tag_matched){
+          res.push({
+            instance_id: instance_stopped['instance_id'],
+            region: instance_stopped['region'],
+            tagKeyValue:(tag_key_value.slice(1)),
+            instance_state: instance_stopped['instance_state'],
+            instance_type: instance_stopped['instance_type']
+          })
+        }
+      })
+	})
+		
+  EOS
+end
+
+###############################################################################
+# Escalations
+###############################################################################
+
+escalation "esc_report_stopped_instances" do
+  email $param_email
+end
+
+escalation "esc_terminate_stopped_instances" do
+  request_approval  do
+    label "Approve Termination of Long-stopped Instances"
+    description "Approve escalation to run RightScale Cloud Workflow to terminate long stopped AWS instances"
+    parameter "approval_reason" do
+      type "string"
+      label "Reason for Approval"
+      description "Explain why you are approving the action"
+    end
+  end
+  run "terminate_instances", data
+end
+
+###############################################################################
+# Policy
+###############################################################################
+
+policy "pol_list_aws_instances" do
+  validate $ds_list_aws_instances do
+    summary_template "{{ rs_project_name }} (Account ID: {{ rs_project_id }}): {{ len data }} AWS Instances Stopped for Over {{parameters.param_stopped_days}} Day(s)"
+    detail_template <<-EOS
+# AWS Instance Details
+| Instance ID | State | Region | Instance Type | Tags |
```

Comment:
I don't think we need state.  this can be removed.
can we short the list by region
If we know the number of days it's been stopped, can we add that column.  If it's too much effort, we can leave it out.

Reply:
1.can we short the list by region >> I did not get this point
2.If we know the number of days it's been stopped, can we add that column. If it's too much effort, we can leave it out >> It looks like too much of effort we have to put to consider number of days particular Instance is stopped

Reply:
@Poojakravi   Sorry, short is a typo.  I'd like to see the data sorted by the region column.  We can skip adding the details for number of days stopped.

Reply:
@cdwilhelm,
I have addressed changes for data sorted by the region column.